core/src/test/java/org/testcontainers/utility/DefaultImageNameSubstitutorTest.java

```diff
@@ -0,0 +1,43 @@
+package org.testcontainers.utility;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.testcontainers.utility.ImageNameSubstitutor.LogWrappedImageNameSubstitutor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class DefaultImageNameSubstitutorTest {
+
+    public static final DockerImageName ORIGINAL_IMAGE = DockerImageName.parse("foo");
+    public static final DockerImageName SUBSTITUTE_IMAGE = DockerImageName.parse("bar");
+    private ConfigurationFileImageNameSubstitutor underTest;
+    private TestcontainersConfiguration mockConfiguration;
+
+    @Before
+    public void setUp() {
+        mockConfiguration = mock(TestcontainersConfiguration.class);
```

Comment:
JFYI we also have `MockTestcontainersConfigurationRule` available :)

Reply:
Updated to use it!